app/models/post_mover.rb

```diff
@@ -365,9 +364,7 @@ def copy_topic_users
              )                                           AS last_emailed_post_number,
              GREATEST(tu.first_visited_at, t.created_at) AS first_visited_at,
              GREATEST(tu.last_visited_at, t.created_at)  AS last_visited_at,
-             CASE
-               WHEN p.user_id IS NOT NULL THEN tu.notification_level
-               ELSE :default_notification_level END      AS notification_level,
+             tu.notification_level,
```

Comment:
doesn't this need deleting? 

```
 LEFT OUTER JOIN
           (
             SELECT DISTINCT user_id
             FROM posts
             WHERE topic_id = :new_topic_id
           ) p ON (p.user_id = tu.user_id)
```

Reply:
I don't think so, it is still used here https://github.com/discourse/discourse/blob/444744dc665d5ada5e58040342eb72ac3221e6fd/app/models/post_mover.rb#L345

Reply:
we can move it back as select instead of join https://github.com/discourse/discourse/commit/2c011252f1c0443965ca5317169550083a43c657#diff-7f70dc8c4811aee4cbe0e70d87254dedL330-L335

Reply:
I think having it in the `SELECT` was a little bit faster than doing the `JOIN`. It might be worth looking at the explain plan of both queries and use the faster one.